src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckCodeFixProvider.cs

```diff
@@ -77,12 +77,20 @@ private static ExpressionSyntax GetCondition(SyntaxNode node)
             var conditionPart = (BinaryExpressionSyntax)conditionLocation.FindNode(cancellationToken);
             var asExpression = (BinaryExpressionSyntax)asExpressionLocation.FindNode(cancellationToken);
 
-            var updatedConditionPart = SyntaxFactory.IsPatternExpression(
+            var updatedConditionPart = (ExpressionSyntax)SyntaxFactory.IsPatternExpression(
                 asExpression.Left, SyntaxFactory.DeclarationPattern(
                     ((TypeSyntax)asExpression.Right).WithoutTrivia(),
                     SyntaxFactory.SingleVariableDesignation(
                         localDeclaration.Declaration.Variables[0].Identifier.WithoutTrivia())));
 
+            // We should negate the is-expression if we have something like "x == null"
+            if (conditionPart.Kind() == SyntaxKind.EqualsExpression)
```

Comment:
:bulb: `IsKind`

Reply:
I use `Kind()` where the code asserts that the variable is never null i.e the path is unreachable if it was. This is also the convention used on the compiler side.

Reply:
I suppose on the IDE side we traditionally use IsKind to not throw if anything ever failed. I'll update all usages.